### PR TITLE
Add Tenant ID to V2 Tracer

### PR DESCRIPTION
**Pull request by vowelparrot**
Update the V2 tracer to
- use UUIDs instead of int's
- load a tenant ID and use that when saving sessions

### CODE REVIEW

File: langchain/callbacks/manager.py [hunk 1]
```diff
@@ -21,14 +21,17 @@
...
+from langchain.callbacks.tracers.schemas import TracerSessionV2
 from langchain.schema import AgentAction, AgentFinish, LLMResult
 
 Callbacks = Optional[Union[List[BaseCallbackHandler], BaseCallbackManager]]
 
 openai_callback_var: ContextVar[Optional[OpenAICallbackHandler]] = ContextVar(
     "openai_callback", default=None
 )
-tracing_callback_var: ContextVar[Optional[LangChainTracer]] = ContextVar(
+tracing_callback_var: ContextVar[
```
Comment (agola11) [inline langchain/callbacks/manager.py:32]: why did this need to be changed? `LangChainTracerV2` is a subclass of `LangChainTracer`

File: langchain/callbacks/tracers/schemas.py [hunk 2]
```diff
@@ -93,7 +105,7 @@ class Run(BaseModel):
     serialized: dict
     inputs: dict
     outputs: Optional[dict]
-    session_id: int
+    session_id: UUID
```
Comment (vowelparrot) [inline langchain/callbacks/tracers/schemas.py:108]: Oops need to fix this

File: langchain/callbacks/tracers/langchain.py [hunk 3]
```diff
@@ -90,13 +101,80 @@ def load_default_session(self) -> TracerSession:
...
+        """Initialize the LangChain tracer."""
+        super().__init__(**kwargs)
+        self._endpoint = _get_endpoint()
+        self._headers = _get_headers()
+        self.tenant_id = _get_tenant_id()
+
+    def _get_session_create(
+        self, name: Optional[str] = None, **kwargs: Any
+    ) -> TracerSessionBase:
+        return TracerSessionV2Base(name=name, extra=kwargs, tenant_id=self.tenant_id)
```
Comment (agola11) [inline langchain/callbacks/tracers/langchain.py:132]: Why is this `TracerSessionV2Base` and not `TracerSessionV2Create`?

File: langchain/callbacks/tracers/schemas.py [hunk 4]
```diff
@@ -31,6 +31,18 @@ class TracerSession(TracerSessionBase):
     id: int
 
 
+class TracerSessionV2Base(TracerSessionBase):
```
Comment (agola11) [inline langchain/callbacks/tracers/schemas.py:34]: make this `TracerSessionV2Create` and it needs an optional UUID

**Review by agola11 — APPROVED** [review]
overall lgtm just a few comments/questions

File: langchain/callbacks/manager.py (hunk 1 above)
Comment (vowelparrot) [reply langchain/callbacks/manager.py:32]: I un-subclassed it but I can go back. Felt a bit more error prone to be overwriting two levels of methods

File: langchain/callbacks/tracers/schemas.py (hunk 4 above)
Comment (vowelparrot) [reply langchain/callbacks/tracers/schemas.py:34]: Are you saying the tenant ID needs to be optional or it should have the `id` key but it's optional here? I'm not sure when we'd want to have it be optional ?

File: langchain/callbacks/tracers/langchain.py (hunk 3 above)
Comment (vowelparrot) [reply langchain/callbacks/tracers/langchain.py:132]: Didn't seem necessary to make a `Create` class before it required unique variables from the `Base` class, but I can mirror the structure of the v1 sessions

**Comment by vowelparrot** [comment]
Ran an LLM and agent trace against the old backend and confirms they're still captured after these changes

File: langchain/callbacks/tracers/langchain.py [hunk 5]
```diff
@@ -81,22 +94,89 @@ def _load_session(self, session_name: Optional[str] = None) -> TracerSession:
...
+            if session_name:
+                params["name"] = session_name
+            r = requests.get(url, headers=self._headers, params=params)
+            tracer_session = TracerSessionV2(**r.json()[0])
+        except Exception as e:
+            session_type = "default" if not session_name else session_name
+            logging.warning(
+                f"Failed to load {session_type} session, using empty session: {e}"
+            )
+            tracer_session = TracerSessionV2(id=1, tenant_id=self.tenant_id)
```
Comment (agola11) [inline langchain/callbacks/tracers/langchain.py:168]: This line is going to break because id is supposed to be a UUID

Comment (vowelparrot) [reply langchain/callbacks/tracers/langchain.py:168]: Ya Forgot to backport from my other PR. Will fix!

File: langchain/callbacks/tracers/langchain.py [hunk 6]
```diff
@@ -81,22 +94,89 @@ def _load_session(self, session_name: Optional[str] = None) -> TracerSession:
...
+        super().__init__(**kwargs)
+        self._endpoint = _get_endpoint()
+        self._headers = _get_headers()
+        self.tenant_id = _get_tenant_id()
+
+    def _get_session_create(
+        self, name: Optional[str] = None, **kwargs: Any
+    ) -> TracerSessionBase:
+        return TracerSessionV2Create(name=name, extra=kwargs, tenant_id=self.tenant_id)
+
```
Comment (vowelparrot) [inline langchain/callbacks/tracers/langchain.py:135]: Would be helpful if we raised proper errors right now for "resource exists". Right now the intention is
- Try to create
- If erros, try to fetch that thing.
- If errors then use an empty session.

**Pull request merged**